### ISSUE TITLE
Plugin not working with basic settings from readme' example

### ISSUE THREAD

**Issue by jesuismaxime** [verbb/knock-knock#53]
### Describe the bug

Got an error with the basic example settings. Seems to comes from empty `allowIps` and `denyIps`. Try to assign an empty array instead of an empty string - and to simply not include those settings, and then the bug is even worse.

Those settings shouldn't be mandatory.

Full stacktrace: 
```
yii\base\ErrorException: Undefined variable $value in /{redacted path}/vendor/verbb/knock-knock/src/models/Settings.php:122
Stack trace:
#0 /{redacted path}/vendor/craftcms/cms/src/web/ErrorHandler.php(77): yii\base\ErrorHandler->handleError(2, 'Undefined varia...', '/{redacted path}/...', 122)
#1 /{redacted path}/vendor/verbb/knock-knock/src/models/Settings.php(122): craft\web\ErrorHandler->handleError(2, 'Undefined varia...', '/{redacted path}/...', 122)
#2 /{redacted path}/vendor/verbb/knock-knock/src/models/Settings.php(62): verbb\knockknock\models\Settings->_getArrayFromMultiline('')
#3 /{redacted path}/vendor/verbb/knock-knock/src/KnockKnock.php(121): verbb\knockknock\models\Settings->getAllowIps()
#4 [internal function]: verbb\knockknock\KnockKnock->verbb\knockknock\{closure}(Object(yii\base\Event))
#5 /{redacted path}/vendor/yiisoft/yii2/base/Component.php(633): call_user_func(Object(Closure), Object(yii\base\Event))
#6 /{redacted path}/vendor/craftcms/cms/src/base/ApplicationTrait.php(1483): yii\base\Component->trigger('init')
#7 /{redacted path}/vendor/craftcms/cms/src/web/Application.php(105): craft\web\Application->_postInit()
#8 /{redacted path}/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#9 /{redacted path}/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct(Array)
#10 [internal function]: yii\base\Application->__construct(Array)
#11 /{redacted path}/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs(Array)
#12 /{redacted path}/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build('craft\\web\\Appli...', Array, Array)
#13 /{redacted path}/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('craft\\web\\Appli...', Array, Array)
#14 /{redacted path}/vendor/craftcms/cms/src/Craft.php(53): yii\BaseYii::createObject(Array, Array)
#15 /{redacted path}/vendor/craftcms/cms/bootstrap/bootstrap.php(239): Craft::createObject(Array)
#16 /{redacted path}/vendor/craftcms/cms/bootstrap/web.php(40): require('/{redacted path}/...')
#17 /{redacted path}/Sites/• starters/tacos-v2-edition-craft-cms/public_html/index.php(11): require('/{redacted path}/...')
#18 {main}
```

### Steps to reproduce

1. Install the plugin
2. Add the settings from the example block available in the readme file

### Craft CMS version

4.1.3

### Plugin version

2.0.1

### Multi-site?

No

**Comment by engram-design** [comment]
Applying the settings from the readme doesn't throw any sort of error for me, because `allowIps` and `denyIps` are empty strings, but yes if they are empty arrays that's invalid.

I've re-worked things to use arrays for these sorts of settings, which they probably always should've been. We still do config normalisation, but at least it's the other way around (string to array).

Should be fixed in [2.0.2](https://github.com/verbb/knock-knock/releases/tag/2.0.2)

**Comment by jesuismaxime** [comment]
- Thank you, now it's working properly (it didn't work at all with empty strings, like the 'old' documentation, as explain earlier).
